client/auth/login.jsx

```diff
@@ -134,47 +163,66 @@ module.exports = React.createClass( {
 							<Gridicon icon="user"/>
 							<FormTextInput
 								name="login"
-								ref="login"
 								disabled={ requires2fa || inProgress }
 								placeholder={ this.translate( 'Username or email address' ) }
 								onFocus={ this.recordFocusEvent( 'Username or email address' ) }
-								valueLink={ this.linkState( 'login' ) } />
+								value={ this.state.login }
+								onChange={ this.updateLogin } />
 						</div>
 						<div className="auth__input-wrapper">
 							<Gridicon icon="lock" />
 							<FormPasswordInput
 								name="password"
-								ref="password"
 								disabled={ requires2fa || inProgress }
 								placeholder={ this.translate( 'Password' ) }
 								onFocus={ this.recordFocusEvent( 'Password' ) }
 								hideToggle={ requires2fa }
 								submitting={ inProgress }
-								valueLink={ this.linkState( 'password' ) } />
+								value={ this.state.password }
+								onChange={ this.updatePassword } />
 						</div>
-						{ requires2fa &&
+						{ ( required2faType === 'code' ) &&
 							<FormFieldset>
 								<FormTextInput
 									name="auth_code"
 									type="number"
-									ref="auth_code"
+									ref={ this.focusWhenIdle }
```

Comment:
It focuses the input when the [input is not in progress](https://github.com/Automattic/wp-calypso/pull/6852/files#diff-3e1c694d3dc270347e2a8aaa73ad1f48R131). I could not think of a more suitable wording to describe "focus when not in progress".